### DISCUSSION TITLE
The user's key seems to be split into three shares. Among them, where is each key share stored particularly?

### DISCUSSION THREAD

**Discussion by yashovardhan** [Web3Auth/Web3Auth#330] [Help]
I see from your documentation that the key is split into 3 shares. It tells what each shares is corresponding to, but doesn't tell where it is stored.
1. Where is the Device Share stored for each device?
2. Where do I get the backup share. Is it safe to have an email backup share?
3. Where does the social share stored?

**Comment by himanshuchawla009** [comment]
- Device share is stored in browser's local storage. In the case of mobile, apps can manage this share in application storage. It is returned to dapp in the dappShare field.

- Backup Share:- This share is sent to the user's email, User can choose to download this as well.  Yes, it is safe to have an email backup share since this is not your primary share which means that losing this share doesn't mean losing your account. You need at least 2 shares to login into your account. Also, you can delete this share from your account management portal: https://app.openlogin.com and can generate a new backup share as well.

- Social login share:- This share is managed by a login service via node operators. The social login share is split into 9 further shares and each of those 9 shares is stored in 9 different nodes. 
You can read about Web3Auth network and nodes here: https://web3auth.io/docs/overview/key-management/#powered-by-the-auth-network

**Reply by irux** [reply]
@himanshuchawla009 I have a question regarding this. Where do I get the email this key? does this only applies for social login access or custom access works like this too? When I log in on my firebase app, I don't get any email with the share. Where can I find this one ?
